### PR TITLE
Use doughnut crate provided sig checks and validation

### PR DESCRIPTION
**Pull request by jordy25519**
Doughnuts now use schnorrkel signing context `b"doughnut"` so need to use the doughnut-rs crate for verification.
- We can remove the old PlugDoughnut verify implementation since new ed/sr signing functions seem to support wasm out of the box.

- PlugDoughnut validation impl should re-use the doughnut crate provided impl

- Some unused test doughnut structs removed

- Tidied up tests alot

### CODE REVIEW

File: primitives/runtime/src/traits.rs [hunk 1]
```diff
@@ -44,92 +44,65 @@ pub use doughnut::{
...
+	fn get_domain(&self, domain: &str) -> Option<&[u8]>;
+	/// Validate the doughnut is usable by a public key (`who`) at the current timestamp (`not_before` <= `now` <= `expiry`)
+	fn validate<Q: AsRef<[u8]>, R: TryInto<u32>>(&self, who: Q, now: R) -> Result<(), ValidationError>;
 }
 
 // Dummy implementation for unit type
 impl PlugDoughnutApi for () {
-    type PublicKey = [u8; 32];
-    type Timestamp = u32;
+	type PublicKey = [u8; 32];
```
Comment (jordy25519) [inline primitives/runtime/src/traits.rs:78]: spaces replaced with tabs here

File: primitives/runtime/src/testing.rs [hunk 2]
```diff
@@ -405,88 +405,3 @@ impl<AccountId, Origin, Call, Extra, Info, Doughnut> Applyable for TestXt<Accoun
 		Ok(res.map_err(Into::into))
 	}
 }
-
-pub mod doughnut {
```
Comment (jordy25519) [inline primitives/runtime/src/testing.rs:409]: unused

File: primitives/runtime/src/traits.rs [hunk 3]
```diff
@@ -44,92 +44,65 @@ pub use doughnut::{
...
-    /// Return the doughnut payload bytes
-    fn payload(&self) -> Vec<u8>;
-    /// Return the doughnut signature
-    fn signature(&self) -> Self::Signature;
-    /// Return the doughnut signature version
-    fn signature_version(&self) -> u8;
-    /// Return the payload for domain, if it exists in the doughnut
-    fn get_domain(&self, domain: &str) -> Option<&[u8]>;
-    /// Validate the doughnut is usable by a public key (`who`) at the current timestamp (`not_before` <= `now` <= `expiry`)
-	fn validate<Q, R>(&self, who: Q, now: R) -> Result<(), ValidationError>
```
Comment (jordy25519) [inline primitives/runtime/src/traits.rs:74]: shouldn't have been re-impl'd here

File: client/peerset/src/lib.rs [hunk 4]
```diff
@@ -592,7 +592,6 @@ mod tests {
 	use futures::prelude::*;
 	use super::{PeersetConfig, Peerset, Message, IncomingIndex, ReputationChange, BANNED_THRESHOLD};
 	use std::{collections::HashSet, pin::Pin, task::Poll, thread, time::Duration};
-	use crate::RESERVED_NODES;
```
Comment (jordy25519) [inline client/peerset/src/lib.rs:595]: unused

File: primitives/runtime/src/testing.rs [hunk 5]
```diff
@@ -460,33 +458,4 @@ pub mod doughnut {
 			write!(f, "TestAccountId({:?})", self.0)
 		}
 	}
-
-	/// A test doughnut
-	#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq)]
```
Comment (jordy25519) [inline primitives/runtime/src/testing.rs:465]: no longer needed

**Review by aliXsed — APPROVED** [review]

**Review by KarishmaBothara — APPROVED** [review]

**Comment by jordy25519** [comment]
had to rebase. will merge once CI passes

**Pull request merged**